File: door/src/store.js
import Vue from 'vue'
import Vuex from 'vuex'

Vue.use(Vuex)

const store = new Vuex.Store({
  state: {
    allow: {},
    auth: null,
    event: null,
    products: null,
    stripeKey: null,
    username: null,
    admitted: 0,
    sold: 0,
    cash: 0,
    check: 0,
  },
  mutations: {
    login(state, { allow, auth, stripeKey, username }) {
      state.allow = allow
      state.auth = auth
      state.stripeKey = stripeKey
      state.username = username
    },
    logout(state) {
      state.auth = state.stripeKey = state.username = state.event = state.products = null
      state.allow = {}
      state.admitted = state.sold = state.cash = state.check = 0
    },
    event(state, { event, products }) {
      state.event = event
      state.products = products
    },
    admitted(state, count) {
      state.admitted += count
    },
    sold(state, { count, amount, method }) {
      state.sold += count
      state.admitted += count
      switch (method) {
        case 'Cash': state.cash += amount; break
        case 'Check': state.check += amount; break;
      }
    },
  },
})
export default store

// TEMPORARY FOR TESTING
// store.commit('login', { auth: '9999-9999-9999', username: 'sroth', allow: { card: true, cash: true, willcall: true }, stripeKey: 'pk_test_QPwvhWbGaakWn7DGcco8J5Nd' })

File: Schola POS/Schola POS/Store.swift
//
//  Store.swift
//  Schola POS
//
//  Created by Steven Roth on 2019-07-23.
//  Copyright © 2019 Schola Cantorum. All rights reserved.
//

import Foundation

struct Allow {
    var card = false
    var cash = false
    var willcall = false
}

struct Event: Codable {
    var id: String
    var name: String
    var start: String
    var freeEntries: [String]?
}

struct Product: Codable {
    var id: String
    var name: String
    var message: String?
    var price: Int!
    var ticketCount: Int!
}

var store = Store()

class Store {

    var username = ""
    var auth = ""
    var allow = Allow()
    var baseURL = ""
    var event: Event!
    var products: [Product] = []
    var logoutCallback: () -> Void = {}
    var admitted = 0
    var sold = 0
    var cash = 0
    var check = 0

    func logout() {
        username = ""
        auth = ""
        allow.card = false
        allow.cash = false
        allow.willcall = false
        baseURL = ""
        event = nil
        products = []
        admitted = 0
        sold = 0
        cash = 0
        check = 0
        cardReaderWatcher.disconnect()
        logoutCallback()
    }
}
